src/core/channel/client_setup.c

```diff
@@ -126,11 +126,9 @@ static void setup_cancel(grpc_transport_setup *sp) {
   if (s->in_alarm) {
     cancel_alarm = 1;
   }
+  gpr_mu_unlock(&s->mu);
   if (--s->refs == 0) {
```

Comment:
I realize this is closed, but just to note:

The point of the duplication here is that it needs to be executed *after* the --s->refs, otherwise we have a race condition. It also needs to be executed *before* destroy_setup, or we destroy the mutex that we have locked.